### PR TITLE
LPS-119526 Nested operation using com.liferay.portal.service.impl.PortletLocalServiceImpl can only be performed in production mode

### PR DESCRIPTION
**Pull request by stsquared99**

### CODE REVIEW

**Comment by stsquared99** [comment]
@Preston-Crary Does this look right to you? Preserve the exception as thrown nested inside the system exception, and then iterate through the causes? The other thought was to always un-nest the cttransactionexception

File: portal-kernel/src/com/liferay/portal/kernel/portlet/LiferayPortlet.java [hunk 1]
```diff
@@ -111,16 +111,23 @@ public void processAction(
 			}
 		}
 		catch (PortletException portletException) {
-			Throwable throwable = portletException.getCause();
+			Throwable throwable = portletException;
 
-			if (throwable instanceof CTTransactionException) {
-				_log.error(throwable, throwable);
+			while ((throwable = throwable.getCause()) != null) {
```
Comment (Preston-Crary) [inline portal-kernel/src/com/liferay/portal/kernel/portlet/LiferayPortlet.java:116]: Cause is never null.  See https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portlet/PortalPreferencesImpl.java#L410 for how to do cause scanning.

**Comment by Preston-Crary** [comment]
Please see my comment and resend.  Thanks!

**Comment by Preston-Crary** [comment]
If it's doable, it's always cleaner to prevent the nesting where it's wrapped and rethrown.

**Comment by stsquared99** [comment]
@Preston-Crary I'll try to do that, but the scanning code is helpful either way.

**Comment by stsquared99** [comment]
Resent: https://github.com/Preston-Crary/liferay-portal/pull/1177

**Pull request closed without merging**